### ISSUE TITLE
NVDA function read entire document do not send braille output on web browser mode to Brltty service

### ISSUE THREAD

**Issue by JanuszChmiel** [nvaccess/nvda#19410]
### Brief summary

When user want to use Braille display connected by computer by using Brltty for Windows. NVDA read entire document only read document by using active speech engine. But lines are not displaied on connected Braille display. I will do my best to track The possible last worked NVDA version, where this feature has worked to help The developers to find An regression if any. May be, that this feature has never allowed users to display automatic readed lines on Braille display.

### Steps to reproduce

1. open www.google.com
2. Press Insert+Down Arrow key.
3. You will hear voice output, automatic line reading will work, but no new lines are displaied on The Braille display. Only first line which have been active in The case when user has activated this function.

### Actual behavior

NVDA read document fluently line by line automatically by using active speech engine. But no new lines are being displaied on The braille display automatically.

### Expected behavior

When user press INSERT+Down Arrow key. NVDA should read line by line automatically and new lines should be also displaied on The connected Braille display.

### NVDA type

installed copy

### NVDA version

2025.3.2 (2025.3.2.53588)

### Windows version

Microsoft Windows Version 25H2 (operating system build  26200.7462)

### Name and version of other software in use when reproducing the issue

Mozilla Firefox 146.0.1 X64 Bit edition language edition CS

### Other information about your system

I AM running Windows 11 Professional X64 Bit edition. I Am using Visionbook from Umax company.

### Does the issue still occur after restarting your computer?

I have restarted my computer and the issue still occurs

### If NVDA add-ons are disabled, is your problem still occurring?

I do not have any NVDA add-ons installed

### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?

I have run the COM Registration Fixing Tool and the issue still occurs

**Comment by seanbudd** [comment]
@JanuszChmiel - are you requesting a feature where braille scrolls automatically?

**Comment by JanuszChmiel** [comment]
NO. I do not want this feature. i want so this feature read The entire document work by The exact way like if user read entire document in Microsoft Word or Notepad. In this case, every line is being correctly printed on The GBraille display. Really. But if I activate read The entire document feature inside some virtual buffer, it read alout, but new lines are not printed on Braille display automatically. Please try this. NVDA support auto scrolling Braille display content excellently. For example. In Balabolka T+memo editable field. If user press F5 key data are being scrolled automatically. So this is perfectly implemented in NVDA algorithms. 
I believe, that Read entire cdocument feature can work by The same way inside virtual bbuffers. So if user open AN WEB page. Gecko, Chrome based browsers, Edge, Seamonkey, etc.

**Comment by seanbudd** [comment]
I'm still confused by this request. NVDA currently doesn't support automatic braille scrolling, but you are saying it currently does.

By new lines, do you mean [paragraph start markers](https://download.nvaccess.org/releases/beta/documentation/userGuide.html#BrailleParagraphStartMarkers)? As we have a feature for that already

**Comment by Emil-18** [comment]
@seanbudd I think they want braille to follow the caret in virtual buffers when using say all, as it does in normal edit controls. This happens because in normal edit controls, braille is updated automaticly when the caret moves. When the caret moves in virtual buffers, braille isn't updated automatically

**Comment by JanuszChmiel** [comment]
Thank you for clarification dear MR Emil-18 . So the question is, how complex and if it would be possible to extend NVDA so it would allow read entire document function to display line by line on Braille device automatically. So it would work like in The Edit controls inside Notepad, Akelpad, Balabolka, ETC. If it is not a bug, it would be possible to redesign this bug report as feature request. Thank you for your patience.

**Comment by Emil-18** [comment]
@JanuszChmiel I think it is a bug. I don't think it would be difficult to fix. I think we just need to update braille in BrowseModeDocumentTreeInterceptor.event_caret

**Comment by irrah68** [comment]
I can't say about BRLTTY, I haven't used it for many years. But with my current Braille display, Braillex EL 80c, I can use the Say all functionality in browse mode so that the Braille display will follow the text while the speech synthesizer is speaking the content of a Web page. But this is only possible when I have chosen these Braille settings in NVDA (which I always have, of course):

Tether Braille: to review
Move system caret when routing review cursor: always

If Braille is tethered automatically or to focus, the Braille display won't follow in browse mode when pressing Insert + Down arrow (Say all). But when Braille is tethered to review, the display will be updated as the text is being read.

**Comment by LeonarddeR** [comment]
This is a duplicate of #3287. Note that #20465 should fix this.
